File: clock-backend/pkg/reqres/leave.go
package reqres

import "github.com/jim/clock-backend/pkg/models"

// CreateLeaveRequest represents the request body for creating a leave
type CreateLeaveRequest struct {
	StartDate string `json:"start_date" validate:"required"`
	EndDate   string `json:"end_date" validate:"required"`
	Note      string `json:"note,omitempty"`
}

// UpdateLeaveRequest represents the request body for updating a leave (admin only)
type UpdateLeaveRequest struct {
	StartDate *string `json:"start_date,omitempty"`
	EndDate   *string `json:"end_date,omitempty"`
	Note      *string `json:"note,omitempty"`
}

// LeaveResponse represents the response for a single leave
type LeaveResponse struct {
	Leave *models.LeaveResponse `json:"leave"`
}

// LeavesResponse represents the response for multiple leaves
type LeavesResponse struct {
	Leaves []*models.LeaveResponse `json:"leaves"`
	Total  int                     `json:"total"`
}

// AdminLeavesResponse represents paginated leaves response for admin
type AdminLeavesResponse struct {
	Leaves  []*models.LeaveWithEmployeeResponse `json:"leaves"`
	Total   int                                 `json:"total"`
	Page    int                                 `json:"page"`
	PerPage int                                 `json:"per_page"`
}

File: clock-frontend/types/leave.ts
export interface Leave {
  id: number;
  employee_id: number;
  start_date: string;
  end_date: string;
  note: string | null;
  days: number;
}

export interface LeaveWithEmployee extends Leave {
  employee_name: string;
  employee_username: string;
}

export interface CreateLeaveRequest {
  start_date: string;
  end_date: string;
  note?: string;
}

export interface UpdateLeaveRequest {
  start_date?: string;
  end_date?: string;
  note?: string;
}

export interface LeaveResponse {
  leave: Leave;
}

export interface LeavesResponse {
  leaves: Leave[];
  total: number;
}

export interface AdminLeavesResponse {
  leaves: LeaveWithEmployee[];
  total: number;
  page: number;
  per_page: number;
}
